src/App.js

```diff
@@ -1,25 +1,43 @@
-import logo from './logo.svg';
-import './App.css';
+import React, {useState} from 'react';
+import {Switch, Route, Redirect} from 'react-router-dom'; 
+import 'bootstrap/dist/css/bootstrap.min.css';
+import './Styles/App.css';
+import Home from './components/Home'; 
+import List from './components/List'; 
+
+//add data 
+import data from './data';
+
+const App = () => {
+  //state
+  const [contacts, setContacts] = useState(data()); 
+
+  const addContact = (contact) => {
+    const newContacts = [...contacts];
+    setContacts([...newContacts, contact]); 
+  }; 
+
+  const editContact = (index, contact) => {
+    const newContacts = [...contacts]; 
+    newContacts[index]= contact;
+    setContacts(newContacts); 
```

Comment:
This could technically be a bit more immutable as you don't _really_ want to replace the object you're editing with a completely new one just in case your edit method doesn't edit all the properties (you wouldn't want to make that assumption).

```js
setContacts(contacts.map((existingContact, i) => {
  if (i !== index) {
    return existingContact
  }
  
  return {
    ...existingContact,
    ...contact
  }
})
```